yaml_cpp_vendor-extras.cmake.in

```diff
@@ -5,21 +5,3 @@ else()
   set(yaml-cpp_DIR "${@PROJECT_NAME@_DIR}/../../../opt/yaml_cpp_vendor/share/cmake/yaml-cpp")
 endif()
 message(STATUS "Setting yaml-cpp_DIR to: '${yaml-cpp_DIR}'")
-
-find_package(yaml-cpp REQUIRED)
-
-set(yaml_cpp_vendor_LIBRARIES ${YAML_CPP_LIBRARIES})
-set(yaml_cpp_vendor_INCLUDE_DIRS ${YAML_CPP_INCLUDE_DIR})
-
-# since yaml-cpp 0.8.0, yaml-cpp supports the yaml-cpp::yaml-cpp target
-if(TARGET yaml-cpp::yaml-cpp)
-  list(APPEND yaml_cpp_vendor_TARGETS yaml-cpp::yaml-cpp)
```

Comment:
Hello @clalancette !
@SteveMacenski pointed out in https://github.com/ros-planning/navigation2/pull/4267 that the yaml_cpp_vendor package should have pulled in the dependencies itself.
Why is this dropped in favor of changing CMakeLists.txt in every downstream packages that need yaml-cpp?

Reply:
> @SteveMacenski pointed out in [ros-planning/navigation2#4267](https://github.com/ros-planning/navigation2/pull/4267) that the yaml_cpp_vendor package should have pulled in the dependencies itself.
> Why is this dropped in favor of changing CMakeLists.txt in every downstream packages that need yaml-cpp?

So there are 2 reasons:

1.  When I originally did this, I couldn't actually get this snippet of code to work.  However, when I started debugging it I realized that...
2. Downstream packages should actually do `find_package(yaml-cpp REQUIRED)`.  That matches what we do in other vendor packages, like [spdlog_vendor](https://github.com/ros2/spdlog_vendor), and things that depend on it (https://github.com/ros2/rcl_logging/blob/128d88e6e4ad6df97405564afbdffa2ebc167814/rcl_logging_spdlog/CMakeLists.txt#L19-L20).  The thinking here is that having a vendor package is a "crutch", and someday in the future we may remove it.  Doing `find_package(spdlog_vendor REQUIRED)` thus should only make the paths available, and the downstream packages should indeed always have to do `find_package(spdlog REQUIRED)` to guard against the day where we no longer provide the vendor package.

That said, I understand that this is a change, so if you need me to open a PR to downstream somewhere, let me know and I can do that.

Reply:
Wait - but why wouldn't the vendor include all the things of the software its vendorizing itself? That seems to break common sense encapsulation, no? If we were to require both, wouldn't it be sensible then that the vendorized packages are only in the package.xml to fetch the compiled resources & then in the CMakeLists we're only finding the main package (ie don't `find_package(*_vendor)`? What's the value that the vendorized package provides from the cmake side of things if we have to include the main package separately?

Reply:
> What's the value that the vendorized package provides from the cmake side of things if we have to include the main package separately?

The value is mostly in the "build-if-needed" part, which we utilize on Windows.  The honest truth is that if we didn't have Windows, most of these vendor packages wouldn't exist and we'd just use them from the system.  And someday in the future when we use conda (or similar) to install our dependencies on Windows, we will remove these vendor packages.

> If we were to require both, wouldn't it be sensible then that the vendorized packages are only in the package.xml to fetch the compiled resources & then in the CMakeLists we're only finding the main package (ie don't `find_package(*_vendor)`?

That would be nice.  I remember we've talked about this before, and there was some reason we couldn't or didn't want to just `find_package(yaml-cpp)` and not the vendor package.  Maybe @cottsay can comment more here.